### PR TITLE
Allow skipped status checks

### PR DESCRIPTION
**Pull request by ellimist**
* Add description
* Add screenshots when changing the UI
* Add unit tests

### References
- Jira link: 

### Risks
- Low/Med/High: thing that could happen

### CODE REVIEW

**Comment by grosser** [comment]
how do skipped statuses happen ?

**Comment by davidreuss** [comment]
They happen when you have a github action workflow with a `if` condition on the job, so it's skipped if it evaluates to false.

In this case, we have different users creating `Deployment` objects through the GitHub Deployment API, and we don't want to run our workflow unless it's a specific user who created it.

In any case, i don't think it should crash on this status as it can clearly happen, and likely will be happening much more with the recent adoption of github actions for implementing all sorts of workflows.

File: test/models/commit_status_test.rb [hunk 1]
```diff
@@ -291,6 +291,12 @@ def stub_checks_api(commit_status: status)
         status.state.must_equal 'error'
       end
 
+      it 'maps skipped status to success' do
```
Comment (grosser) [inline test/models/commit_status_test.rb:294]: FYI test should explain why something happens, not what happens
```
it "does not fail when action was intentionally skipped"
```

**Comment by grosser** [comment]
I can roll this out tomorrow

**Comment by ronan-mch** [comment]
@davidreuss - do you expect that further use of GitHub Actions will lead to more status check types? I'm wondering if we're going to need to keep on making this change as we add more Actions...

**Comment by davidreuss** [comment]
@ronan-mch I don't know. But i'm not expecting it, but there must be some documentation on the types of statuses that the github app sets for github action statuses (whew, that was hard to re-read)

**Pull request merged**